### PR TITLE
connect calendar logic to itinerary page and render sights on corresponding dates

### PR DESCRIPTION
**Pull request by heleah**
helloooo, this time I have added the logic from the calendar to the Itinerary page. on selecting a date for a specific sight, the sight will get rendered on the Itinerary page under the corresponding date. if several sights get added to one date, the date only gets rendered once with all sights below. you can also get to the sight's details view from there and delete it. the dates are shown in chronological order.
please note: you can add one sight to the itinerary more than once. on clicking on the delete icon, only that one entry will be deleted.
review link: https://adventeire-feat-itinera-rtkz3g.herokuapp.com/
please check in mobile view only!!
thanks a mill (: ✨ ☘️

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -17,7 +17,10 @@ function App() {
     loadFromLocal('Favourite Sights') ?? []
   );
   const [showCalendar, setShowCalendar] = useState(false);
-  const [selectDate, setSelectDate] = useState('');
+  const [selectDate, setSelectDate] = useState('28/6/2021');
```
Comment (sacidcelik) [inline client/src/App.js:20]: Is the placeholder date going to stay? :) Or is there a specific reason for a fixed date in the initial state?

File: client/src/components/Calendar.js [hunk 2]
```diff
@@ -60,26 +61,8 @@ export default function Calendar({
...
-    ]); */
-    //}
-    let allCombos = dateSightCombos;
-    let newCombo = { date: selectDate, sight: sight.name };
-    allCombos.push(newCombo);
-    setDateSightCombos(...dateSightCombos, allCombos);
-
-    console.log(selectDate, sight.name, dateSightCombos);
+    const newCombo = { date: selectDate, sight: sight };
+    setDateSightCombos([...dateSightCombos, newCombo]);
```
Comment (sacidcelik) [inline client/src/components/Calendar.js:65]: wasn't aware this works :) I would have pushed the value to App.js and made the state change there... but I guess if it works like that? :)

File: client/src/pages/Itinerary.js [hunk 3]
```diff
@@ -1,5 +1,115 @@
...
+  }, [dateSightCombos]);
+
+  function removeFromItinerary(comboToRemove) {
+    const remaining = dateSightCombos.filter(
+      (combo) =>
+        (combo.sight._id && combo.date) !==
+        (comboToRemove.sight._id && comboToRemove.date)
+    );
+    setDateSightCombos(remaining);
+  }
```
Comment (sacidcelik) [inline client/src/pages/Itinerary.js]: When I have more than one sight on a single day in my calendar and want to remove one of them, it deletes all of them. I would assume it has to do with this function?

**Review by sacidcelik — CHANGES_REQUESTED** [review]
Very nice feature! Love that it groups items by date!

Two notes:

- When there is more than 1 sight per date, deleting one results in all being deleted (for that date).
- I am sure you are aware that people can select dates in the past and put them on their itinerary. Might be an intentional choice to allow people to track their sights at a later stage but just wanted to mention it :)

File: client/src/App.js (hunk 1 above)
Comment (heleah) [reply client/src/App.js:20]: Without a placeholder date, the date for the sight that gets selected first won't get picked up for whatever reason. if you have a better idea, please let me know (:

File: client/src/components/Calendar.js (hunk 2 above)
Comment (heleah) [reply client/src/components/Calendar.js:65]: works like a charm 😎

**Pull request merged**